### PR TITLE
Doc: Add documentation for Groovy DSL Image Configuration options

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Add Documentation for Groovy DSL Image configuration options

Related to #808 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_groovy.adoc [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                // ...
+                ports = ['8080']
+            }
+        }
+    }
+}
+----
+
+
+You can read more about supported fields in `image` configuration element in <<config-image, Image Configuration>> section.
```
Comment (manusa) [inline gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_groovy.adoc:185]: The idea was to provide a thorough (numbered) image(s) configuration, and then describe what each point does.
For each point the description should link to the k8sBuild task Configuration section

Comment (manusa) [reply gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_groovy.adoc:185]: Maybe for the assembly part, an additional example is interesting, but individualized examples for every single configuration point is too much.

**Comment by manusa** [comment]
Documentation should be structured in a useful way

`2.1 Groovy configuration` is fine, but shouldn't be so thorough, a few sections with a single example (with numbered items and light descriptions with links) should be enough. These sections could be: Image Configuration (maybe provide an extra section for assemblies), Resource configuration (for future PRs), etc.

Section 3 and 4 should probably be merged, let's leave the grouped structure for now (but maybe we can remove the groups in the future, if they don't add value):
```
3. Tasks Overview
    3.1 Outer loop (Or Build and deployment)
        3.1.1 k8sBuild
    3.2 Inner loop (Or Development)
        3.2.1. k8sDebug
```

Detailed docs such as ImageConfiguration, etc. should have at least some example of Groovy DSL on how to configure or maybe link to the CommonScenario/Task overview sections so that users know how to define the configuration.

Final Sections:
```
1. Introduction
2. Common scenarios
3. Tasks Overview
4. Groovy configuration (thorough and detailed explanation of everything, rest of the doc sections should link to parts within this section)
```

File: gradle-plugin/doc/src/main/asciidoc/inc/build/_overview.adoc [hunk 2]
```diff
@@ -78,20 +78,20 @@ When only a single image should be built with a Dockerfile no Groovy DSL configu
 All what need to be done is to place a `Dockerfile` into the top-level module directory, alongside to `build.gradle`.
 You can still configure <<global-configuration, global aspects>> in the plugin configuration, but as soon as you add an `image` in the groovy DSL configuration, you need to configure also the build explicitly.
 
-The image name is by default set from the Gradle coordinates (`%g/%a:%l`, see <<image-name,Image Name >> for an explanation of the params which are essentially the Maven GAV)
+The image name is by default set from the Gradle coordinates (`%g/%a:%l`, see <<image-name,Image Name >> for an explanation of the params which are essentially the Gradle GAV)
```
Comment (manusa) [inline gradle-plugin/doc/src/main/asciidoc/inc/build/_overview.adoc:81]: I don't thin there is such thing as `Gradle coordinates` or `Maven GAV`

File: gradle-plugin/doc/src/main/asciidoc/inc/_groovy-configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+If the effective build mode is `openshift` an additional `s2i` option is available
+and selected by default.
+
+Available strategies for OpenShift are:
+
+* `s2i` for a https://docs.openshift.com/enterprise/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image build] with a binary source
+* `docker` for a https://docs.openshift.com/enterprise/latest/architecture/core_concepts/builds_and_image_streams.html#docker-build[Docker build] with a binary source
+
+endif::[]
+| `jkube.build.strategy`
```
Comment (rohanKanojia) [inline gradle-plugin/doc/src/main/asciidoc/inc/_groovy-configuration.adoc:25]: Right now these properties don't seem to be working. If I pass them in gradle goal like this, it has no effect:

`gradle -Djkube.build.strategy=docker k8sBuild`

I see it's not handled in KubernetesExtension. Do we want to support this or shall I remove this column from here?

**Review by manusa — APPROVED** [review]

**Pull request merged**